packages/devtools_app/lib/src/config_specific/theme_overrides/theme_overrides_web.dart

```diff
@@ -0,0 +1,81 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be found
+// in the LICENSE file.
+
+import 'dart:html';
+
+import 'package:flutter/widgets.dart';
+
+/// Environment-specific theme overrides, for example IDE-provided
+/// theming for embedded mode.
+class ThemeOverrides {
+  ThemeOverrides({this.backgroundColor, this.foregroundColor, this.fontSize});
+
+  factory ThemeOverrides.load() {
+    final queryParams = _loadQueryParams();
+
+    final overrides = ThemeOverrides(
+      backgroundColor: _tryParseColor(queryParams['backgroundColor']),
+      foregroundColor: _tryParseColor(queryParams['foregroundColor']),
+      fontSize: _tryParseDouble(queryParams['fontSize']),
+    );
+
+    // If the environment has provided a background color, set it immediately
+    // to avoid a white page until the first Flutter frame is rendered.
+    if (overrides.backgroundColor != null) {
+      document.body.style.backgroundColor =
+          overrides.backgroundColor.value.toRadixString(16).padLeft(8, '0');
+    }
+
+    return overrides;
+  }
+
+  Color backgroundColor;
+  Color foregroundColor;
+  double fontSize;
+
+  static Map<String, String> _loadQueryParams() {
+    // The querystring could be in one of two places:
+    // http://localhost:123/?backgroundColor=00000
+    // http://localhost:123/#/?backgroundColor=00000
```

Comment:
should it be considered a bug that we support both?  it would be nice to just support standard url query parameters.

Reply:
That's how I originally did it, but it seems that when the main method ran Flutter had already redirected us to the second. I thought it would be best to support both in case anything changed in Flutter that might affect this (it doesn't seem like parsing the raw querystring like this is a particularly Flutter-like thing to do, so I wasn't sure if we should rely on exactly how it behaved).

If we just want one, the second one is probably the right one. WDYT?